### PR TITLE
SOLR-16962: Restore ability to configure tlog directory

### PR DESCRIPTION
**Pull request by magibney**
backported from in-process upsteam PR at commit ea8e829d78d207b4a6e48f4514737bf6f717c7d4

also includes revert of backport of earlier iteration of same upstream PR

### CODE REVIEW

File: solr/core/src/test-files/solr/configsets/cloud-hdfs/conf/solrconfig.xml [hunk 1]
```diff
@@ -35,7 +35,9 @@
     <commitWithin>
       <softCommit>${solr.commitwithin.softcommit:true}</softCommit>
     </commitWithin>
-    <updateLog></updateLog>
+    <updateLog>
```
Comment (hiteshk25) [inline solr/core/src/test-files/solr/configsets/cloud-hdfs/conf/solrconfig.xml:38]: @magibney this satisfies our usecase test?

**Review by hiteshk25 — APPROVED** [review]
LGTM (Thanks @magibney. Please see if you can our usecase explicitly )

**Comment by magibney** [comment]
merging this to `branch_9_3`; upstream PR is almost complete.

File: solr/core/src/test-files/solr/configsets/cloud-hdfs/conf/solrconfig.xml (hunk 1 above)
Comment (magibney) [reply solr/core/src/test-files/solr/configsets/cloud-hdfs/conf/solrconfig.xml:38]: This (hdfs) is actually irrelevant to our use case. It's even not directly relevant to the PR (which doesn't really exercise explicit `solr.ulog.dir` configuration for hdfs), but it puts a hook in the config file to make such testing possible. This PR _does_ test regular local filesystem explicit `solr.ulog.dir`, which is our usecase (and the most common usecase generally, by far).

**Pull request merged**